app.vue

```diff
@@ -1,21 +1,48 @@
 <script lang="ts" setup>
 import { useAuth0 } from "@auth0/auth0-vue";
 
+const { $appInsights } = useNuxtApp();
+
 const { isLoading, loginWithRedirect, isAuthenticated, error, logout } =
   useAuth0();
 
 watch(
   isLoading,
   async (loading) => {
     if (loading) return;
+
     if (!isAuthenticated.value) {
+      $appInsights.event("auth0 - redirect to login", {});
       await loginWithRedirect();
     }
   },
   { immediate: true },
 );
 
+watch(error, async () => {
+  const e = error.value;
+  if (e) {
+    const errorCode = (e as any).error;
+    console.error(e.message, "\n\n\n", errorCode);
+    $appInsights.event("auth0 - error", {
+      error: errorCode,
+      message: e.message,
+      stack: e.stack,
+    });
+    if (errorCode === "missing_refresh_token") {
+      // For these errors we've identified that a logout & relogin likely will fix it.
```

Comment:
Don't you want this to be logged differently in this case, since we have a way of solving it?

Reply:
I'm not sure if it solves the issue. I first need some real world data. It solved it when I manually provoked the issue on my machine. But there's a good chance it doesn't work in other circumstances.

To get good logging requires an iterative approach. I'll probably create 6-10 small PRs with very small changes over the next 3 months to get this to a good state.